### DISCUSSION TITLE
Mermaid support?

### DISCUSSION THREAD

**Discussion by gforcada** [networkx/networkx#6693] [Ideas]
Hi all!

Thanks for the amazing library! 🤩 

We are using it on https://github.com/collective/pipforester/ to discover cyclic dependencies within python distributions 🚀 

We want to create a GitHub Action out of `pipforester` and display the cyclic dependencies found in [GitHub Action job summaries](https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/).

Unfortunately GitHub does not allow to embed graphs in `DOT` format but rather in [mermaid](https://mermaid.ink/).

Is there any work towards supporting other output formats?

I could try to do that given some direction within the codebase 😄 🍀

**Comment by rossbar** [comment]
AFAIK there is no current effort to support converting graphs->mermaid format within NX. I'd start by checking whether there exists any projects that support converting DOT->mermaid or tikz->mermaid. IMO this would be a better tool as `.dot` format is the defacto text-based format for describing graph visualizations.

**Reply by gforcada** [reply]
I couldn't find any python tool that could do that unfortunately :-/

**Reply by dschult** [reply]
This might be something -- but I haven't looked at it other than the landing page.
https://github.com/utensil/mermaid2dot

**Reply by MridulS** [reply]
FWIW, I personally wouldn't be too against of adding mermaid support to networkx itself, especially because it has core support in Github Markdown and it's big in the js world.

**Reply by dschult** [reply]
The mermaid2dot hasn't been changed since 2015. So it might be better to include something in NetworkX (especially if you are willing to help maintain it for the next year or two. :)  

Within the codebase, this would be included in the `readwrite` folder probably makes sense to have a new module called `mermaid.py` with accompanying tests in `tests/test_mermaid.py`.  Other questions -- just ask...
